src/pj_apply_vgridshift.c

```diff
@@ -212,14 +220,75 @@ int pj_apply_vgridshift( PJ *defn, const char *listname,
                 else
                     sprintf( gridlist+strlen(gridlist), ",%s", gi->gridname );
             }
-            pj_log( defn->ctx, PJ_LOG_DEBUG_MAJOR,
-                    "%s", gridlist );
-                
+
+            proj_log_debug(defn, "%s", gridlist);
             pj_ctx_set_errno( defn->ctx, PJD_ERR_GRID_AREA );
+
             return PJD_ERR_GRID_AREA;
         }
     }
 
     return 0;
 }
 
+/**********************************************/
+int proj_vgrid_init(PJ* P, const char *grids) {
+/**********************************************
+
+  Initizalize and populate gridlist.
+
+    Takes a PJ-object and the plus-parameter
+    name that is used in the proj-string to
+    specify the grids to load, e.g. "+grids".
+    The + should be left out here.
+
+    Returns the number of loaded grids.
+
+***********************************************/
+
+    /* prepend "s" to the "grids" string to allow usage with pj_param */
+    char *sgrids = (char *) pj_malloc( (strlen(grids)+1) *sizeof(char) );
```

Comment:
you need strlen(grids)+1+1 for the 's' and nul terminator

Reply:
I'll take care of that. Thanks for catching that.